java/src/Ice/src/main/java/com/zeroc/Ice/SysLoggerI.java

```diff
@@ -10,69 +10,18 @@
 import java.net.InetAddress;
 
 public final class SysLoggerI implements Logger {
-    public SysLoggerI(String prefix, String facilityString) {
-        int facility;
-        if (facilityString.equals("LOG_KERN")) {
-            facility = LOG_KERN;
-        } else if (facilityString.equals("LOG_USER")) {
-            facility = LOG_USER;
-        } else if (facilityString.equals("LOG_MAIL")) {
-            facility = LOG_MAIL;
-        } else if (facilityString.equals("LOG_DAEMON")) {
-            facility = LOG_DAEMON;
-        } else if (facilityString.equals("LOG_AUTH")) {
-            facility = LOG_AUTH;
-        } else if (facilityString.equals("LOG_SYSLOG")) {
-            facility = LOG_SYSLOG;
-        } else if (facilityString.equals("LOG_LPR")) {
-            facility = LOG_LPR;
-        } else if (facilityString.equals("LOG_NEWS")) {
-            facility = LOG_NEWS;
-        } else if (facilityString.equals("LOG_UUCP")) {
-            facility = LOG_UUCP;
-        } else if (facilityString.equals("LOG_CRON")) {
-            facility = LOG_CRON;
-        } else if (facilityString.equals("LOG_AUTHPRIV")) {
-            facility = LOG_AUTHPRIV;
-        } else if (facilityString.equals("LOG_FTP")) {
-            facility = LOG_FTP;
-        } else if (facilityString.equals("LOG_LOCAL0")) {
-            facility = LOG_LOCAL0;
-        } else if (facilityString.equals("LOG_LOCAL1")) {
-            facility = LOG_LOCAL1;
-        } else if (facilityString.equals("LOG_LOCAL2")) {
-            facility = LOG_LOCAL2;
-        } else if (facilityString.equals("LOG_LOCAL3")) {
-            facility = LOG_LOCAL3;
-        } else if (facilityString.equals("LOG_LOCAL4")) {
-            facility = LOG_LOCAL4;
-        } else if (facilityString.equals("LOG_LOCAL5")) {
-            facility = LOG_LOCAL5;
-        } else if (facilityString.equals("LOG_LOCAL6")) {
-            facility = LOG_LOCAL6;
-        } else if (facilityString.equals("LOG_LOCAL7")) {
-            facility = LOG_LOCAL7;
-        } else {
-            throw new InitializationException(
-                    "Invalid value for Ice.SyslogFacility: " + facilityString);
-        }
-        initialize(prefix, facility);
-    }
-
-    private SysLoggerI(String prefix, int facility) {
-        initialize(prefix, facility);
+    public SysLoggerI(String prefix, String facilityString, String host, int port) {
+        this(prefix, parseFacility(facilityString), parseHost(host), port);
     }
 
-    private void initialize(String prefix, int facility) {
+    private SysLoggerI(String prefix, int facilityString, InetAddress host, int port) {
```

Comment:
`int facilityString` looks dubious to me here.
What is this variable representing?

Reply:
Renamed to facility, it is correspond to the syslog numerical facility.